### PR TITLE
[Design2-125] DSCO - Tabs - check and update the new styles, add light and dark mode

### PR DESCRIPTION
**Pull request by levadadenys**
### [Design2-125] DSCO - Tabs - check and update the new styles, add light and dark mode

* added `light` and `dark` mode support
* added `mode` prop to `Tabs` component

Before:
Знімок екрана 2024-07-10 о 15 12 04
Знімок екрана 2024-07-10 о 15 13 33

After:
Знімок екрана 2024-07-10 о 16 43 06
Знімок екрана 2024-07-10 о 16 43 02

https://github.com/code-dot-org/code-dot-org/assets/22244040/62c7d860-6f24-4f7e-bae8-2de8f29d4884

## Links
- jira ticket: [Design2-125](https://codedotorg.atlassian.net/browse/DESIGN2-125)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/componentLibrary/tabs/tabs.module.scss [hunk 1]
```diff
@@ -35,85 +48,201 @@
...
-      border: none !important;
-      border-bottom: 2px solid $light_primary_500 !important;
+    li button {
+      color: $light_gray_800;
+
+
+      &:hover {
+        color: $light_primary_500;
+        border: none;
+        border-bottom: 2px solid $light_primary_500;
```
Comment (fisher-alice) [inline apps/src/componentLibrary/tabs/tabs.module.scss:67]: On hover, I see that the bottom border doesn't quite extend all the way to the left and right.
Screenshot 2024-07-10 at 1 25 11 PM

As a comparison, here's what it looks like without bottom border on hover. I think maybe this is a question for @markabarrett ?

Screenshot 2024-07-10 at 1 30 20 PM

File: apps/src/componentLibrary/tabs/tabs.module.scss [hunk 2]
```diff
@@ -149,7 +278,7 @@
         @include label-one;
         margin: 0;
         padding: 0 0.75rem;
-        gap: 0.4375rem;
+        gap: 0.375rem;
         height: 2rem;
 
         &:has(i:first-child) {
```
Comment (fisher-alice) [inline apps/src/componentLibrary/tabs/tabs.module.scss:295]: Wondering if we could maybe vertically center the tab text a bit more...Could update `span` margin top to a slightly bigger value (e.g. line 167) to `margin: 3px 0 0;`.

**Comment by levadadenys** [comment]
Fixed border-bottom issue, fixed vertical alignment of text for secondary tabs

**Review by fisher-alice — APPROVED** [review]
Nice updates!

**Pull request merged**